powersimdata/scenario/execute.py

```diff
@@ -187,110 +158,31 @@ def _check_if_ready(self):
             None, which translates to gurobi
-        :return: (*requests.Response*) -- the http response object
-        """
-        scenario_id = self._scenario_id()
-        url = f"http://{server_setup.SERVER_ADDRESS}:5000/launch/{scenario_id}"
-        resp = requests.post(url, params={"threads": threads, "solver": solver})
-        if resp.status_code != 200:
-            print(
-                f"Failed to launch simulation: status={resp.status_code}. See response for details"
-            )
-        return resp
-
-    def _check_threads(self, threads):
-        """Validate threads argument
-
-        :param int threads: the number of threads to be used
-        :raises TypeError: if threads is not an int
-        :raises ValueError: if threads is not a positive value
-        """
-        if threads:
-            if not isinstance(threads, int):
-                raise TypeError("threads must be an int")
-            if threads < 1:
-                raise ValueError("threads must be a positive value")
-
-    def _check_solver(self, solver):
-        """Validate solver argument
-
-        :param str solver: the solver used for the optimization
-        :raises ValueError: if invalid solver provided
-        """
-        solvers = ("gurobi", "glpk")
-        if solver is not None and solver.lower() not in solvers:
-            raise ValueError(f"Invalid solver: options are {solvers}")
-
-    def launch_simulation(self, threads=None, extract_data=True, solver=None):
-        """Launches simulation on target environment (server or container)
-
-        :param int/None threads: the number of threads to be used. This defaults to None,
-            where None means auto.
         :param bool extract_data: whether the results of the simulation engine should
             automatically extracted after the simulation has run. This defaults to True.
-        :param str solver: the solver used for optimization. This defaults to
-            None, which translates to gurobi
         :return: (*subprocess.Popen*) or (*requests.Response*) - either the
             process (if using ssh to server) or http response (if run in container)
+            or (*dict*) (if run locally)
         """
         self._check_if_ready()
-        self._check_threads(threads)
-        self._check_solver(solver)
 
         mode = get_deployment_mode()
         print(f"--> Launching simulation on {mode.lower()}")
-        if mode == DeploymentMode.Server:
-            return self._launch_on_server(threads, solver, extract_data)
-        return self._launch_in_container(threads, solver)
+        return self._launcher.launch_simulation(threads, solver, extract_data)
 
     def check_progress(self):
-        """Get the lastest information from the server container
+        """Get the status of an ongoing simulation, if possible
 
-        :raises NotImplementedError: if not running in container mode
+        :return: (*dict*) -- progress information, or None
```

Comment:
Maybe we should elaborate the returns for different launchers, SSH, HTTP, native?